packages/benchmarks/src/textToDriver/makeDatabases/loadBraintrustDatasets.ts

```diff
@@ -0,0 +1,108 @@
+import { initDataset } from "braintrust";
+import { EJSON } from "bson";
+import {
+  CollationOptions,
+  CreateIndexesOptions,
+  Document,
+  IndexDirection,
+  IndexSpecification,
+} from "mongodb-rag-core";
+
+interface LoadBraintrustDatasetParams {
+  apiKey: string;
+  projectName: string;
+  datasetName?: string;
+}
+
+export async function loadBraintrustMetadata({
+  apiKey,
+  projectName,
+  datasetName = "database-metadata",
+}: LoadBraintrustDatasetParams) {
+  const dataset = initDataset(projectName, {
+    apiKey,
+    dataset: datasetName,
+  });
+  const metadata: DatabaseMetadata[] =
+    // TODO: why do i have to cast this as unknown?
+    (
+      (await dataset.fetchedData()) as unknown as DatabaseMetadataDatasetEntry[]
+    ).map(
+      (m) =>
+        ({
+          databaseName: m.input.databaseName,
+          collections: m.input.collections.map((c) => ({
+            collectionName: c.collectionName,
+            indexes: c.indexes.map((idx) =>
+              EJSON.deserialize(idx)
+            ) as unknown as CollectionMetadata["indexes"],
+          })),
+        } satisfies DatabaseMetadata)
+    );
+  return metadata;
+}
+
+export interface DatabaseMetadataDatasetEntry {
+  input: {
+    dataset: string;
+    databaseName: string;
+    collections: {
+      collectionName: string;
+      indexes: Record<string, unknown>[];
+    }[];
+  };
+}
+
+export interface DatabaseMetadata {
+  databaseName: string;
+  collections: CollectionMetadata[];
+}
+
+export interface CollectionMetadata {
+  collectionName: string;
+  indexes: {
+    v: CreateIndexesOptions["version"];
+    key: Map<string, IndexDirection>;
+    name: CreateIndexesOptions["name"];
+    // collation?: CollationOptions;
+  }[];
+}
+
+export async function loadBraintrustDbDocuments({
+  apiKey,
+  projectName,
+  datasetName = "database-documents",
+}: LoadBraintrustDatasetParams) {
+  const dataset = initDataset(projectName, {
+    apiKey,
+    dataset: datasetName,
+  });
+  const dbDocuments: DbDocument[] =
+    // TODO: why do i have to cast this as unknown?
```

Comment:
I pulled the PR down and don't get a type error if I remove the unknown, i.e this builds successfully for me:

```ts
((await dataset.fetchedData()) as DatabaseMetadataDatasetEntry[]).map(...)
```

Still sucks that we need the type cast. I think it might be solved with some better generics in the braintrust lib. In the meantime, we could avoid this cast altogether by parsing with `zod` instead.

Reply:
>  I think it might be solved with some better generics in the braintrust lib

agree. they're generally pretty good about supporting generics. surprised they don't have it for this method. will 2x check to make sure that the generic isn't supported and file a ticket for them if it in fact is not

> we could avoid this cast altogether by parsing with zod instead.

yea, good point. let me throw that together

Reply:
the removing cast as unknown is also working for me..not sure what was up before. was getting funky ts err. 

anyways, converted to zod types and using that. 

aside: gh copilot is magic at taking a TS type and creating the zod type for it.

Reply:
created issue for braintrust https://github.com/braintrustdata/braintrust-sdk/issues/384